### PR TITLE
#1385 new view sub as parent

### PR DESCRIPTION
**Pull request by magicmilo**
This is work to allow parent accounts to view their subsidiary accounts as normal

#### Work done
- parentSubsidiaryViewService
- Pages to replace each tab in subsidiary view
- Routing service to intercept component links to their respective subsidiary page
- Breadcrumb journeys for new pages
- Breadcrumb code to act in subsidiary with existing journeys
- Removed subsidiary tabs 
- subsidiary account component
- banner code and events

#### Tests
Does this PR include tests for the changes introduced?
- [ ] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -26,5 +26,9 @@
   "[scss]": {
     "editor.formatOnSave": true,
     "editor.defaultFormatter": "esbenp.prettier-vscode"
-  }
+  },
+  "search.exclude": {
+    "**.spec.ts": true,
+    "**/.angular/**": true,
+  },
```
Comment (duncanc19) [inline .vscode/settings.json:33]: Remove

File: frontend/src/app/core/components/subsidiaryAccount/subsidiaryAccount.component.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+@import 'govuk-frontend/govuk/base';
+
+.asc-tabs-container {
+  background-color: #e6e5e3;
+}
+
+.asc-navigation-container {
+  background-color: govuk-colour('light-grey');
+  border-bottom: 1px solid #e6e5e3;
+}
```
Comment (duncanc19) [inline frontend/src/app/core/components/subsidiaryAccount/subsidiaryAccount.component.scss:10]: Could maybe move into shared scss file?

File: frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    private trainingCategoryService: TrainingCategoryService,
+    private trainingService: TrainingService,
+  ) {}
+
+  ngOnInit(): void {
+    this.establishmentService.setCheckCQCDetailsBanner(false);
+    this.breadcrumbService.show(JourneyType.SUBSIDIARY);
+    this.subsidiaryUid = this.route.snapshot.params['subsidiaryUid'];
+
+    // this.workplace = this.route.snapshot.data.subsidiaryWorkplaceResolver;
```
Comment (duncanc19) [inline frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts:63]: Remove

File: frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        .pipe(take(1))
+        .subscribe((trainingCategories) => {
+          this.trainingCategories = trainingCategories;
+        }),
+    );
+  }
+
+  public navigateToStaffRecords(event: Event): void {
+    event.preventDefault();
+    // this.workerService.tabChanged.next(true);
```
Comment (duncanc19) [inline frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts:125]: Remove?

File: frontend/src/app/features/workplace/edit-workplace/edit-workplace.component.ts [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
-  constructor(private route: ActivatedRoute, private establishmentService: EstablishmentService) {}
+  constructor(
+    private route: ActivatedRoute,
+    private establishmentService: EstablishmentService,
+    private parentSubsidiaryViewService: ParentSubsidiaryViewService
+  ) {}
 
   ngOnInit() {
-    this.establishmentService.setState(this.route.snapshot.data.establishment);
+    // this.establishmentService.setState(this.route.snapshot.data.establishment); // TODO
```
Comment (duncanc19) [inline frontend/src/app/features/workplace/edit-workplace/edit-workplace.component.ts:18]: Is this needed?

File: frontend/src/app/features/workplace/start/start.component.html [hunk 6]
```diff
@@ -67,7 +67,7 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-8 govuk__nowrap">Add more detai
       <li><p class="govuk-!-margin-left-3 govuk-!-margin-bottom-0">whether we can share your data?</p></li>
     </ul>
     <a
-      [routerLink]="['/workplace', establishment.uid, 'other-services']"
+      [routerLink]="['/subsidiary/workplace', establishment.uid, 'other-services']"
```
Comment (duncanc19) [inline frontend/src/app/features/workplace/start/start.component.html:70]: Will this never be used by a standalone?

File: frontend/src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts [hunk 7]
```diff
@@ -154,7 +157,8 @@ export class WorkplaceInfoPanelComponent implements OnInit, OnDestroy {
         this.establishmentService.setEmployerTypeHasValue(false);
         this.router.navigate(['/workplace', this.workplace.uid, 'type-of-employer']);
       } else {
-        this.router.navigate(['/workplace', this.workplace.uid]);
+        this.parentSubsidiaryViewService.setViewingSubAsParent(this.workplace.uid);
+        this.router.navigate(['/subsidiary/home/', this.workplace.uid]);
```
Comment (duncanc19) [inline frontend/src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts:161]: Not too clear what this is doing

File: frontend/src/app/features/workplace/workplace-name-address/workplace-name-address.component.ts [hunk 8]
```diff
@@ -12,11 +12,13 @@ import { WorkplaceNameAddressDirective } from '@shared/directives/create-workpla
...
 @Component({
   selector: 'app-workplace-name-address',
-  templateUrl:
-    '../../../shared/directives/create-workplace/workplace-name-address/workplace-name-address.component.html',
+  templateUrl: '../../../shared/directives/create-workplace/workplace-name-address/workplace-name-address.component.html'
 })
+
 export class WorkplaceNameAddressComponent extends WorkplaceNameAddressDirective {
   public workplace: Establishment;
+  public link: string = '/workplace';
```
Comment (duncanc19) [inline frontend/src/app/features/workplace/workplace-name-address/workplace-name-address.component.ts:20]: Doesn't look like this is being used

File: frontend/src/app/shared/components/back-to-parent-link/back-to-parent-link.component.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    this.subscriptions.add(
+      this.establishmentService.getEstablishment(this.parentUid, true).subscribe((workplace) => {
+        this.establishmentService.setState(workplace);
+      }),
+    );
+  }
+
+  public backToParentLinkClick(event: Event) {
+    event.preventDefault();
+    //this.setWorkplace();
```
Comment (duncanc19) [inline frontend/src/app/shared/components/back-to-parent-link/back-to-parent-link.component.ts:36]: Remove

File: frontend/src/app/shared/components/data-area-tab/about-the-data-link/about-the-data-link.component.ts [hunk 10]
```diff
@@ -13,17 +14,16 @@ export class AboutTheDataLinkComponent implements OnInit {
...
     this.workplaceUid = this.establishmentService ? this.establishmentService.primaryWorkplace.uid : null;
+
   }
 
   public setReturn(): void {
-    this.benchmarksService.setReturnTo({
-      url: [this.router.url.split('#')[0]],
-      fragment: 'benchmarks',
-    });
+    this.router.navigate(['/workplace', this.workplaceUid, 'data-area', 'about-the-data']);
```
Comment (duncanc19) [inline frontend/src/app/shared/components/data-area-tab/about-the-data-link/about-the-data-link.component.ts:27]: Doesn't seem to be setting a return, just navigating. May need renaming?

File: frontend/src/app/shared/components/new-workplace-summary/workplace-summary.component.ts [hunk 11]
```diff
@@ -37,13 +37,16 @@ export class NewWorkplaceSummaryComponent implements OnInit, OnDestroy {
   public numberOfStaffWarning: boolean;
   public typeOfEmployer: string;
 
+  public link: string = '/workplace';  // TODO REMOVE! REVERT!
```
Comment (duncanc19) [inline frontend/src/app/shared/components/new-workplace-summary/workplace-summary.component.ts:40]: Sounds like we want to revert to just having /workplace in each link for the html file?

File: frontend/src/app/shared/directives/new-home-tab/new-home-tab.directive.ts [hunk 12]
```diff
@@ -221,7 +224,9 @@ export class NewHomeTabDirective implements OnInit, OnDestroy {
     this.canViewReports =
       this.permissionsService.can(workplaceUid, 'canViewWdfReport') ||
       this.permissionsService.can(workplaceUid, 'canRunLocalAuthorityReport');
-    this.canViewEstablishment = this.permissionsService.can(workplaceUid, 'canViewEstablishment');
+    //this.canViewEstablishment = this.permissionsService.can(workplaceUid, 'canViewEstablishment');
+
+    this.canViewEstablishment = true;
```
Comment (duncanc19) [inline frontend/src/app/shared/directives/new-home-tab/new-home-tab.directive.ts:229]: Is this always going to be true? Either way can remove the commented out code

File: frontend/src/app/features/subsidiary/benchmarks/view-subsidiary-benchmarks.component.html [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+          [viewBenchmarksComparisonGroups]="viewBenchmarksComparisonGroups"
+          [showRegisteredNurseSalary]="showRegisteredNurseSalary"
+        ></app-data-area-pay>
+      </ng-template>
+    </div>
+    <!-- <app-data-area-about-the-data
+      #aboutData
+      class="govuk-visually-hidden"
+      [workplace]="workplace"
+    ></app-data-area-about-the-data> -->
```
Comment (duncanc19) [inline frontend/src/app/features/subsidiary/benchmarks/view-subsidiary-benchmarks.component.html:102]: Do we not need to show this here?

File: frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+  public totalExpiringTraining: number;
+  public missingMandatoryTraining: number;
+  public staffMissingMandatoryTraining: number;
+  public totalStaff: number;
+  public isShowAllTrainings: boolean;
+  public viewTrainingByCategory = false;
+  public staffSortByValue = 'trainingExpired';
+  public trainingSortByValue = '0_expired';
+  public canEditWorker = true; // TODO
+  public canEditEstablishment = true; // TODO
```
Comment (duncanc19) [inline frontend/src/app/features/subsidiary/training-and-qualifications/view-subsidiary-training-and-qualifications.component.ts:43]: Need to update these to use permissions service

File: frontend/src/app/shared/components/back-to-parent-link/back-to-parent-link.component.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+  ngOnInit() {}
+
+  private setWorkplace(): void {
+    this.subscriptions.add(
+      this.establishmentService.getEstablishment(this.parentUid, true).subscribe((workplace) => {
+        this.establishmentService.setState(workplace);
+      }),
+    );
+  }
```
Comment (duncanc19) [inline frontend/src/app/shared/components/back-to-parent-link/back-to-parent-link.component.ts:32]: This function isn't being used, looks like we don't need it if we're just routing back to dashboard when going back to the parent

File: Makefile [hunk 16]
```diff
@@ -4,6 +4,7 @@
 all: install run
 
 install:
+	export NODE_ENV=localhost
```
Comment (jpmaHub) [inline Makefile:7]: remove this since we don't need this anymore to run make commands

**Review by duncanc19 — APPROVED** [review]

**Pull request merged**